source/NVDAObjects/UIA/__init__.py

```diff
@@ -1832,6 +1832,24 @@ def event_UIA_notification(self, notificationKind=None, notificationProcessing=U
 				speech.cancelSpeech()
 			ui.message(displayString)
 
+	def event_UIA_dragComplete(self):
+		"""
+		Raised by an element when a drag operation on it is completed.
+		A prominent example is rearranging Start screen tiles on Windows 8 and later.
+		"""
+		# #12271: speech does not report new location but braille does.
+		self.reportFocus()
+
+	def event_UIA_droptTargetDropped(self):
+		"""
+		Raised by a containing element such as a list when an element is dragged over another element.
+		A prominent example is rearranging quick actions in Windows 10 Action Center.
+		"""
+		# #12271: unlike drag complete event, it is a different element that raises this event.
+		# but NVDA does record the newly dragged element as focused object.
+		api.getFocusObject().reportFocus()

```

Comment:
What if the dragged element is entirely unrelated to the focused object?

Reply:
That's a distinct possibility. We may need to find more places where UIA drag/drop events are fired - so far, prominent places I found are ones I noted in code comments.